src/nstime.c

```diff
@@ -141,8 +141,12 @@ nstime_monotonic_t *nstime_monotonic = JEMALLOC_N(n_nstime_monotonic);
 #endif
 bool
 nstime_update(nstime_t *time) {
-	nstime_t old_time;
+	if (nstime_monotonic()) {
+		nstime_get(time);
+		return false;
+	}
 
+	nstime_t old_time;
 	nstime_copy(&old_time, time);
 	nstime_get(time);
 
```

Comment:
The input time can be set by the caller, so in principle this function needs to check for time going backward even if the clock is monotonic, unless we require the caller to never pass in a computed (possibly future) time.

Is there an issue you ran into while working on lock profiling that caused you to experiment with this change?